fs/read_json.ts

```diff
@@ -1,7 +1,15 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 
+export type JSONPrimitive = string | number | boolean | null;
+export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
+export interface JSONObject {
+  [member: string]: JSONValue;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface JSONArray extends Array<JSONValue> {}
```

Comment:
Why can’t this use a `type` instead of an `interface`?

Reply:
A good question.

Reply:
Because if I use `type`

```typescript
export type JSONPrimitive = string | number | boolean | null;
export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
export interface JSONObject {
  [member: string]: JSONValue;
}
export type JSONArray = Array<JSONValue>
```

 ts will report an error.

> Type alias 'JSONValue' circularly references itself.ts(2456)

an `ts error` or `eslint error`. I choose the latter.